packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx

```diff
@@ -34,16 +36,32 @@ class ArchiveAudioPlayer extends Component {
    * Check if Track index has changed,
    * if so, then play that track
    */
-  componentDidUpdate({ sourceData: { index: prevIndex } }, { playerPlaylistIndex: prevPlaylistIndex }) {
-    const { player, playerPlaylistIndex } = this.state;
-    const { sourceData: { index } } = this.props;
+  componentDidUpdate(prevProps, prevState) {
+    const { player, playerPlaylistIndex, initiatePlay } = this.state;
+    const { sourceData: { index = null } } = this.props;
+
+    // if state index doesn't match props index, manually play
+    const indexHasBeenSet = Number.isInteger(index);
+
+    // if there is an index in props, but flag is false,
+    // set flag to true;
+    let stateUpdate = {};
+    let postStateCB = null;
+    if (indexHasBeenSet && !initiatePlay) {
+      stateUpdate.initiatePlay = true;
+    }
 
-    const propsIndexChanged = prevIndex !== index;
-    const playerIndexChanged = prevPlaylistIndex !== playerPlaylistIndex;
-    const manuallyJumpToTrack = propsIndexChanged && !playerIndexChanged;
+    const manuallyJumpToTrack = (playerPlaylistIndex !== index) && indexHasBeenSet && initiatePlay;
 
     if (manuallyJumpToTrack) {
-      return player.playN(index);
+      stateUpdate.playerPlaylistIndex = index;
+      postStateCB = () => {
+        player.playN(index);
+      }
+    }
+
+    if (Object.keys(stateUpdate).length) {
```

Comment:
Optional change but AirBNB JS advocates for explicit comparisons for strings and numbers.